fix theseus

TingquanGao opened the pull request:
unify the pattern of layer's name

TingquanGao committed:
fix: unify the pattern of layer's name

TingquanGao committed:
fix: fix result returned by _find_layers_handle

TingquanGao committed:
fix: fix result returned by stop_after

TingquanGao committed:
fix: fix comments

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
-            pattern_list = return_pattern.split(".")
+    def _return_dict_hook(self, layer, input, output):
+        res_dict = {"output": output}
+        for res_key in list(self.res_dict):
```
不需要写 list()

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+        for pattern in patterns:
+            pattern_list = pattern.split(".")
             if not pattern_list:
                 continue
```
可能是多余的了

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
-            sub_layer.res_dict = self.res_dict
-            sub_layer.res_name = return_pattern
-            sub_layer.register_forward_post_hook(sub_layer._save_sub_res_hook)
+            sub_layer = getattr(sub_layer_parent, sub_layer_name, False)
```
用None做默认值更好

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+            try:
+                sub_layer = sub_layer[
+                    sub_layer_index] if sub_layer_index is not None else sub_layer
+            except KeyError as e:
```
可以试试len()

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+            else:
+                setattr(sub_layer_parent, sub_layer_name, sub_layer)
+
+            handle_res = handle_func(sub_layer, pattern)
```
handle提前

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+                sub_layer = wrap_theseus(sub_layer)
+
+            if sub_layer_index:
+                getattr(sub_layer_parent,
```
直接修改layer

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+    def _save_sub_res_hook(self, layer, input, output):
+        self.res_dict[self.res_name] = output
+
+    def _find_layers_handle(self, patterns, handle_func):
```
两个handle不能表达不同的意思

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+                layer.res_dict = self.res_dict
+                layer.res_name = pattern
+                layer.register_forward_post_hook(layer._save_sub_res_hook)
+                return True
```
return layer

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
-
-        '''
+            if not isinstance(sub_layer, TheseusLayer):
+                sub_layer = wrap_theseus(sub_layer)
```
不需要wrap

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+        return stop_after
+
+    def replace_sub(self,
+                    layer_name_pattern: Union[str, List[str]],
```
与find_layers_handle合并

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,206 @@ def __init__(self, *args, **kwargs):
+        return self._find_layers_handle(
+            layer_name_pattern, handle_func=replace_function)
+
+    # TODO(weishengyu): stop doesn't work when stop layer has a parallel branch.
```
太麻烦，不做了……

TingquanGao committed:
fix: fix problems commented in reviewing

TingquanGao committed:
refactor: extract _parse_pattern_str() func

TingquanGao committed:
refactor: strengthen parse_pattern_str() func

TingquanGao committed:
refactor: rename replace_sub() func to ?

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,141 @@ def __init__(self, *args, **kwargs):
+        raise DeprecationWarning(msg)
+
+    # TODO(gaotingquan): what is a good name?
+    def layer_wrench(self,
```
?

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -19,99 +19,141 @@ def __init__(self, *args, **kwargs):
+    def _save_sub_res_hook(self, layer, input, output):
+        self.res_dict[self.res_name] = output
+
+    def replace_sub(self, *args, **kwargs) -> None:
```
需要更新PaddleClas种调用了这个函数的代码

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -124,5 +166,144 @@ def forward(self, *inputs, **kwargs):
+            "target_layer": target_layer,
+            "target_layer_name": target_layer_name,
+            "target_layer_index": target_layer_index,
+            "parent_layer": parent_layer
```
yield->list

weisy11 commented on ppcls/arch/backbone/base/theseus_layer.py:
```diff
@@ -124,5 +166,144 @@ def forward(self, *inputs, **kwargs):
+        Iterator[Dict[str, Union[nn.Layer, None, str]]]: Dict["target_layer": Union[nn.Layer, None], "target_layer_name": str, "target_layer_index": str, "parent_layer": nn.Layer]
+    """
+
+    pattern_list = slice_pattern(pattern, idx)
```
!

TingquanGao committed:
refactor: simplify code

1. remove WrapLayer and wrap_theseus;
2. support call update_res() one more;
3. optim parse_pattern_str() to return list of layer parsed.

TingquanGao committed:
add license

TingquanGao committed:
fix: remove unnecessary register_hook() call & pre-commit

TingquanGao committed:
fix: rename to upgrade_sublayer()

TingquanGao committed:
fix: replace reg exp with layer name

weisy11 reviewed (approved):
LGTM
